routing/turns_tts_text.cpp

```diff
@@ -45,10 +46,195 @@ void GetTtsText::ForTestingSetLocaleWithJson(std::string const & jsonBuffer, std
   m_getCurLang = platform::ForTestingGetTextByIdFactory(jsonBuffer, locale);
 }
 
+// Returns -1 for not found, positive for which item in the haystack was found
+long FindInStrArray(const std::vector<std::string>& haystack, std::string& needle)
+{
+  auto begin = haystack.begin();
+  for(std::vector<std::string>::const_iterator i = begin; i != haystack.end(); ++i) {
+    std::string item = *i;
+    std::string::size_type idx = needle.find(item);
+    if (idx != std::string::npos) {
+      long item_i = std::distance( begin, i );
+      LOG(LINFO, ("TTSf found", item_i, item, idx, needle));
+      return item_i;
+    }
+  }
+  return -1;
+}
+
+void HungarianBaseWordTransform(std::string& myString) {
+  std::string_view baseStr[] = {"e", "a", "ö", "ü"};
+  std::string_view harmonyStr[] = {"é", "á", "ő", "ű"};
+
+  for (size_t i = 0; i < std::size(baseStr); ++i)
+  {
+    if (strings::EndsWith(myString, baseStr[i]))
+      myString.replace(myString.size() - baseStr[i].size(), baseStr[i].size(), harmonyStr[i]);
+  }
+}
+
+uint8_t CategorizeHungarianAcronymsAndNumbers(std::string const & myString)
+{
+  std::vector<std::string> backNames = {
+      "A", // a
+      "Á", // á
+      "H", // há
+      "I", // i
+      "Í", // í
+      "K", // ká
+      "O", // o
+      "Ó", // ó
+      "U", // u
+      "Ű", // ú
+      "0", // nulla or zéró
+      "3", // három
+      "6", // hat
+      "8", // nyolc
+  };
+
+  std::vector<std::string> frontNames = {
+      // all other letters besides H and K
+      "B", "C", "D", "E", "É", "F", "G", "J", "L", "M", "N", "Ö", "Ő", "P", "Q", "R", "S", "T", "Ú", "Ü", "V", "W", "X", "Y", "Z",
+      "1", // egy
+      "2", // kettő
+      "4", // négy
+      "5", // öt
+      "7", // hét
+      "9", // kilenc
+  };
+
+  std::vector<std::string> specialCaseFront = {
+      "10", // tíz special case front
+      "40", // negyven front
+      "50", // ötven front
+      "70", // hetven front
+      "90", // kilencven front
+  };
+
+  std::vector<std::string> specialCaseBack = {
+      "20", // húsz back
+      "30", // harminc back
+      "60", // hatvan back
+      "80", // nyolcvan back
+  };
+
+  //'100', // száz back
+
+  strings::UniString original = strings::MakeUniString(myString);
+  for (std::string::size_type i = myString.size()-1; i > 0; i--) {
+    // special is 2 char, so check last 2
+    std::string oneBuf = strings::ToUtf8(strings::Substring(original, i, 1));
+    std::string twoBuf = strings::ToUtf8(strings::Substring(original, i-1, 2));
+    std::string threeBuf = strings::ToUtf8(strings::Substring(original, i-2, 3));
+    if (FindInStrArray(specialCaseFront, twoBuf) != -1) {
+      return 1;
+    }
+    if (FindInStrArray(specialCaseBack, twoBuf) != -1) {
+      return 2;
+    }
```

Comment:
```suggestion
    if (FindInStrArray(specialCaseBack, twoBuf) != -1)
      return 2;
```
Here and everywhere.